pubsub.go

```diff
@@ -72,6 +75,18 @@ type PubSub struct {
 	// a notification channel for when our peers die
 	peerDead chan peer.ID
 
+	// tracks the channels subscribed to new peer notifications
+	newPeerNotif map[chan peer.ID]struct{}
+
+	// setup new peer notifier
+	addPeerNotif chan *reqNewPeerNotifier
```

Comment:
what's this?

Reply:
The pubsub discovery module is asking to be notified of which peer streams we've connected to/failed to connect to. We need this to know when we're done trying to bootstrap. There may also be other uses to having a channel tracking new peers (currently we have `newPeerNotif map[chan peer.ID]struct{}` but it could easily be `newPeerNotif map[chan {peer.ID, Subscribed|Disconnected|etc.}]struct{}`

Reply:
we don't care about this in the mainloop.